### PR TITLE
Use explicit styling for Javascript

### PR DESCRIPTION
**Pull request by Chlodochar**
This PR partly follows #2748 (Added Javascript styling rules).

## Description
At the moment the JS files were a bit of a mess:

* Mixed tabs and spaces (on same line! My eyes!)
* Mixed naming conventions (camel-case, usage of _, etc)
* Different usage of spaces before and after keywords
* Undefined variables (... and implicitly define them in `window`)

To circumvent this, a style guide needs to be chosen or created. To get things going, I chose [StandardJS](http://standardjs.com/). I don't care which style-guide we'll use, but it has to be explicit, creating a style-guide ourselves is hard and a lot of work. StandardJS seems to have enough tools available to check and format code. So that's why I chose it.

In addition I added eslint configuration, so that javascript can be checked. Lastly I tried to automatically format map.js and app.js. I've made a script to do this, so rebases are still doable. There are still a number of open issues that eslint reports, but that can be fixed after the general style is applied (manual work, not something I'll like to rebase).

See the resulting map.js here:
https://github.com/Chlodochar/PokemonGo-Map/blob/pr-jsstyle/static/map.js

## Motivation and Context
Code style was a mess. To get a unified code-base, some style-guide needs to be used.

## How Has This Been Tested?
Go through some of the features in the webinterface.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ? ] My code follows the code style of this project. (which one?!)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Chlodochar** [comment]
Related to #1806. This will also allow usage of Travis or SideCI.

**Comment by CarlTaylor1989** [comment]
I'm getting issues with this. Can you have a look?

screen shot 2016-07-26 at 23 11 00

**Comment by Chlodochar** [comment]
@CarlTaylor1989 Could you give this another try?

**Comment by Chlodochar** [comment]
I've now only ran only safe changes on the code. At the moment it will not rename variables. That does leave some errors left, but those can be changed manually one the general styling of the code is ok.

**Comment by invisiblek** [comment]
✖ 230 problems (230 errors, 0 warnings)

Are you still working on this? Otherwise I'll jump on it and finish it up

**Comment by Chlodochar** [comment]
Ah thanks. I'll work on the leftover problems.

File: static/js/map.js [hunk 1]
```diff
@@ -1,233 +1,804 @@
...
+  'stylers': [{
+    'visibility': 'off'
+  }]
+}, {
+  'featureType': 'all',
+  'elementType': 'labels.icon',
+  'stylers': [{
+    'visibility': 'off'
+  }]
+}]
```
Comment (andrewda) [inline static/js/map.js]: Might want these to go in separate files or stay one-liners. I don't think the styles are a great use of about 600 lines.

Comment (Chlodochar) [inline static/js/map.js]: agreed. however, we also do not want lines that are 1000+ characters long.

**Comment by Chlodochar** [comment]
No linting errors left. All features that I've found seem to be working like they should. There is still a problem in `/stats` relating to jQuery Accordion not being included, but that was incorrect before this PR.

**Pull request closed without merging**